soft_delete_sample.py

```diff
@@ -264,86 +246,61 @@ def deleted_certificate_recovery(self):
-        print('recovered certificate {}\n{}'.format(cert_to_recover, self._serialize(cert)))
+        recovered_certificate_poller = certificate_client.begin_recover_deleted_certificate(cert_to_recover)
+        recovered_certificate_certificate = recovered_certificate_poller.result()
+        print('recovered certificate {}'.format(recovered_certificate_certificate.name))
 
         # purge a deleted certificate
-        self.keyvault_data_client.purge_deleted_certificate(vault.properties.vault_uri, cert_to_purge)
+        certificate_client.purge_deleted_certificate(cert_to_purge)
+        time.sleep(50)
         print('purged certificate {}'.format(cert_to_purge))
 
-        # list the vaults certificate
-        keys = self.keyvault_data_client.get_certificates(vault.properties.vault_uri)
-        print('certificates: \n{}'.format(self._serialize(certs)))
-
-    def _wait_on_delete_completed(self, vault_uri, entity_type, entity_name):
-        if entity_type == 'secret':
-            get_deleted_func = self.keyvault_data_client.get_deleted_secret
-            args = (vault_uri, entity_name)
-        elif entity_type == 'key':
-            get_deleted_func = self.keyvault_data_client.get_deleted_key
-            args = (vault_uri, entity_name)
-        elif entity_type == 'certificate':
-            get_deleted_func = self.keyvault_data_client.get_deleted_certificate
-            args = (vault_uri, entity_name)
-        else:  # entity_type == 'vault'
-            get_deleted_func = self.keyvault_mgmt_client.vaults.get_deleted
-            args = (entity_name, self.config.location)
-        self._poll_while_404(get_deleted_func, args)
-
-    def _wait_on_recover_completed(self, vault_uri, entity_type, entity_name):
-        if entity_type == 'secret':
-            get_func = self.keyvault_data_client.get_secret
-            args = (vault_uri, entity_name, KeyVaultId.version_none)
-        elif entity_type == 'key':
-            get_func = self.keyvault_data_client.get_key
-            args = (vault_uri, entity_name, KeyVaultId.version_none)
-        elif entity_type == 'certificate':
-            get_func = self.keyvault_data_client.get_certificate
-            args = (vault_uri, entity_name, KeyVaultId.version_none)
-        else:  # entity_type == 'vault'
-            get_func = self.keyvault_mgmt_client.vaults.get
-            args = (entity_name, self.config.location)
-        self._poll_while_404(get_func, args)
-        
-    def _poll_while_404(self, func, args=(), retry_wait=10, max_retries=10):
-        for x in range(max_retries):
-            try:
-                return func(*args[0:]) 
-            except KeyVaultErrorException as e:
-                if not e.response.status_code == 404:
-                    raise e
-                time.sleep(retry_wait)
+        # list the vault certificates
+        certificates = certificate_client.list_properties_of_certificates()
+        print('list the vault certificates')
```

Comment:
To align with print statements from the key and secret samples:
```suggestion
        print('all of the certificates in the client's vault:')
```